core/domain/collection_jobs_one_off.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for collections."""
+
+import logging
+
+from core import jobs
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.platform import models
+import feconf
+
+(base_models, collection_models,) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.collection])
+
+
+class CollectionMigrationJobManager(jobs.BaseMapReduceJobManager):
+    """A reusable one-time job that may be used to migrate collection schema
+    versions. This job will load all existing collections from the data store
+    and immediately store them back into the data store. The loading process of
+    a collection in collection_services automatically performs schema updating.
+    This job persists that conversion work, keeping collections up-to-date and
+    improving the load time of new collections.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [collection_models.CollectionModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        # Note: the read will bring the collection up to the newest version.
+        collection = collection_services.get_collection_by_id(item.id)
+        try:
+            collection.validate()
+        except Exception as e:
+            logging.error(
+                'Collection %s failed non-strict validation: %s' %
+                (item.id, e))
+            return
+
+        # Write the new collection into the datastore.
+        commit_cmds = [{
+            'cmd': collection_domain.CMD_MIGRATE_SCHEMA,
+            'version': str(
+                feconf.CURRENT_COLLECTION_SCHEMA_VERSION)
+        }]
+        collection_services.update_collection(
```

Comment:
See the comments above; the collection version gets updated when a collection is loaded from model, so all collection objects in the backend already have the latest schema version, so I don't think avoiding the update can be done unless we want to change when the schema version is updated.